ytfzf

```diff
@@ -1134,13 +1142,76 @@ get_history () {
 }
 
 clear_history () {
-	if [ -e "$history_file" ]; then
-		: > "$history_file"
-		printf "History has been cleared\n"
+    case "$1" in
+        search) h_file="${search_history_file}" ;;
+        watch) h_file="${history_file}" ;;
+        *) h_file="${search_history_file}|${history_file}" ;;
+    esac
+
+    exit_code=0
+    IFS="|"
+    for file in $h_file; do
+        if [ "$file" = "$history_file" ]; then
+            name="History"
+        else
+            name="Search History"
+        fi
+        if [ -e "$file" ]; then
+            : > "$file"
+            printf "%s has been cleared\n" "$name"
+        else
+            printf "\033[31m%s file not found, history not cleared\033[0m\n" "$name" >&2
+            exit_code=1
+        fi
+    done
+    exit $exit_code
+}
+
+get_search_history () {
+	if [ "$enable_search_hist" -eq 1 ]; then
+		[ -e "$search_history_file" ] || touch "$search_history_file"
+		#gets history data in reverse order (makes it most recent to least recent)
+		hist_data=$( sed '1!G; h; $!d' "$search_history_file" )
+		[ -z "$hist_data" ] && printf "Search history is empty!\n" >&2 && return 1;
+		#removes duplicate values from $history_data
+		search_history=$(printf "%s" "$hist_data" | uniq )
 	else
-		printf "\033[31mHistory file not found, history not cleared\033[0m\n" >&2
-		exit 1
+		printf "Search history is not enabled. Please enable it to use this option (-q).\n";
+		exit;
 	fi
+	unset hist_data
+}
+
+set_search_history () {
+    [ -z "$search_query" ] && return
+    [ $enable_search_hist -eq 1 ] && printf "%s\t%s\n" "$(date '+%Y-%m-%d %H:%M:%S')" "$search_query" >> "$search_history_file" ;
+}
+
+search_history_menu () {
+    get_search_history
+
+    if [ "$is_ext_menu" -eq 1 ]; then
+    #when using an external menu, the search history will be done there
+        choice=$( printf "%s\n" "$search_history" | eval "$external_menu" )
+    else
+        choice="$( printf "%s\n" "$search_history" | fzf --print-query --no-multi -d '\t' --with-nth=2.. --expect='alt-enter' --bind='tab:replace-query' )"
+    fi
+
+    query="$( printf "%s" "$choice" | sed -n '1p' )"
+    [ -z "$query" ] && exit
```

Comment:
From testing, I think this line shouldn't exist, when i dont use an ext menu, it just exits here.

Reply:
That line is important to make sure that ytfzf exits when nothing has been selected, without it you can have some odd behavior.

Reply:
hmm, well something needs to be changed i think because it seems to exit even though i'm selecting something.

Reply:
Ahh I figured it out, it needs to only exit if both query and search is empty.

Reply:
instead if checking if query is empty, maybe after query, key and search are defined you can use this line 
```sh
[ -z "$query" ] && [ -z "$key" ] && [ -z "$search" ] && exit
```

also when using an ext_menu, instead of
```sh
choice=$(printf "%s\n" "$search_query" | eval "$external_menu")
```
you could change choice to search_query, and then return on the next line.